port/common/omrfile.c

```diff
@@ -607,41 +607,33 @@ EsTranslateOpenFlags(int32_t flags)
 void
 omrfile_vprintf(struct OMRPortLibrary *portLibrary, intptr_t fd, const char *format, va_list args)
 {
-	char outputBuffer[256];
-	char *allocatedBuffer;
-	uint32_t numberWritten;
-	va_list copyOfArgs;
-
-	/* Attempt to write output to stack buffer */
-	COPY_VA_LIST(copyOfArgs, args);
-	numberWritten = portLibrary->str_vprintf(portLibrary, outputBuffer, sizeof(outputBuffer), format, copyOfArgs);
-
-	/* str_vprintf always null terminates, returns number characters written excluding the null terminator */
-	if (sizeof(outputBuffer) > (numberWritten + 1)) {
-		/* write out the buffer */
-		portLibrary->file_write_text(portLibrary, fd, outputBuffer, numberWritten);
-		return;
-	}
+	char localBuffer[256];
+	char *writeBuffer = NULL;
 
-	/* Either the buffer was too small, or it was the exact size.  Unfortunately can't tell the difference,
```

Comment:
You will need to bring back all of the COPY_VA_LIST code.  See other comments about this.